bucket-policy-handlers.go

```diff
@@ -0,0 +1,269 @@
+/*
+ * Minio Cloud Storage, (C) 2015, 2016 Minio, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package main
+
+import (
+	"bytes"
+	"crypto/sha256"
+	"encoding/hex"
+	"io"
+	"io/ioutil"
+	"net/http"
+	"net/url"
+	"regexp"
+	"strings"
+
+	mux "github.com/gorilla/mux"
+	"github.com/minio/minio/pkg/fs"
+	"github.com/minio/minio/pkg/probe"
+	"github.com/minio/minio/pkg/s3/access"
+)
+
+// maximum supported access policy size.
+const maxAccessPolicySize = 20 * 1024 * 1024 // 20KiB.
+
+// isActionValid - verify if action is valid.
+func isActionValid(action string, u *url.URL, statements []accesspolicy.Statement) bool {
```

Comment:
checkBucketPolicyAction - Verify if this "action" is allowed in the bucket policy for corresponding resource.